picoquic/sender.c

```diff
@@ -402,7 +403,8 @@ uint32_t picoquic_protect_packet(picoquic_cnx_t* cnx,
     picoquic_connection_id_t * local_cnxid,
     uint32_t length, uint32_t header_length,
     uint8_t* send_buffer, uint32_t send_buffer_max,
-    void * aead_context, void* pn_enc)
+    void * aead_context, void* pn_enc,
+    int contains_retrans)
 {
```

Comment:
This is wrong. We go to great length to ensure that retransmission is a "frame level" issue, not a "packet level" issue. You may find that the current code does that, but it can change at any time. For example, instead of triggering an actual packet retransmission, the code could simply add the frames from the lost packet to the "misc frames" queue. In fact, we will have to do that if PMTU discover that the packet size is now shorter than expected. There will not be any retransmitted packet, but instead the lost stream frame will be split in two "misc" frames, retransmitted in independent packets.